core/parser.mly

```diff
@@ -1013,14 +968,11 @@ binding:
 | LINFUN var arg_lists block                                   { with_pos $loc (`Fun (make_untyped_binder $2,
                                                                                   `Lin, ([], ($3, with_pos $loc($4) (`Block $4))), `Unknown,
                                                                                   None)) }
-| typedecl SEMICOLON                                           { $1 }
 | typed_handler_binding                                        { with_pos $loc (`Handler $1) }
-| links_module                                                 { $1 }
-| alien_block                                                  { $1 }
-| links_open                                                   { $1 }
+| typedecl SEMICOLON | links_module | alien_block | links_open { $1 }
 
 bindings:
-| binding                                                      { [$1] }
+| binding                                                      { [$1]      }
 | bindings binding                                             { $1 @ [$2] }
```

Comment:
shouldn't this collapse to a single  rule  ` bindings -> binding+`  ?

(sorry, the original comment was on the wrong line)

Reply:
I will reply in main discussion - this is too long for an inline comment.